### PR TITLE
Claim bug fix

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-1918

## What changed (plus any additional context for devs)
The problem:
Grant enabled claimables on TF, and then downloaded prod. He then claimed a few claimables. The claims were successful, however the claimables data displayed on the wallet screen failed to update, and never updated even after a day of waiting and restarting the app.

Why it happened:
The way feature flags are accessed on the wallet screen is different than everywhere else in the app, which can cause partial enablement of features when switching from TF to prod. This is what happened with claimables. The claimables feature SHOULD have been automatically disabled when switching to prod, however it was still enabled on the wallet screen while the claimables query was disabled (which caused the data on the wallet screen to never update).

The solution:
I fixed the way feature flags values are accessed on the wallet screen. This change guarantees that in prod the feature flag values will always be the default ones, not the ones leftover from TF.
Other changes:
* added a query `refetchInterval` of 2 min
* immediately remove a claimable from the query data if the claim was successful
* invalidate (and refetch) claimables query data 20s after claim functionality completes, regardless of outcome. this will repopulate a claimable if incorrectly removed, or remove a claimable if it was incorrectly displayed (which may have led to a failed claim outcome) 

## What to test
set condition on line 17 of experimentalHooks.ts to false and make sure feature flags work properly

### CODE REVIEW

File: src/screens/claimables/ClaimingSponsoredClaimable.tsx [hunk 1]
```diff
@@ -86,6 +90,10 @@ export const ClaimingSponsoredClaimable = ({ claimable }: { claimable: Sponsored
         );
       }
     },
+    onSettled: () => {
+      // Clear and refresh claimables data 30s after claim button is pressed, regardless of success or failure
+      delay(() => queryClient.invalidateQueries(queryKey), 20_000);
+    },
```
Comment (walmat) [inline src/screens/claimables/ClaimingSponsoredClaimable.tsx:96]: 20s or 30s?

File: src/screens/claimables/ClaimingClaimableSharedUI.tsx [hunk 2]
```diff
@@ -146,7 +146,7 @@ export const ClaimingClaimableSharedUI = ({
   });
 
   const onPress = useCallback(
-    debounce(() => {
+    debounce(async () => {
```
Comment (walmat) [inline src/screens/claimables/ClaimingClaimableSharedUI.tsx:149]: does this need to be async?

**Review by walmat — APPROVED** [review]

File: src/screens/claimables/ClaimingClaimableSharedUI.tsx (hunk 2 above)
Comment (benisgold) [reply src/screens/claimables/ClaimingClaimableSharedUI.tsx:149]: naur

File: src/screens/claimables/ClaimingSponsoredClaimable.tsx (hunk 1 above)
Comment (benisgold) [reply src/screens/claimables/ClaimingSponsoredClaimable.tsx:96]: eyes of an eagle

File: src/screens/claimables/ClaimingSponsoredClaimable.tsx [hunk 3]
```diff
@@ -86,6 +90,10 @@ export const ClaimingSponsoredClaimable = ({ claimable }: { claimable: Sponsored
         );
       }
     },
+    onSettled: () => {
+      // Clear and refresh claimables data 20s after claim button is pressed, regardless of success or failure
+      delay(() => queryClient.invalidateQueries(queryKey), 20_000);
```
Comment (greg-schrammel) [inline src/screens/claimables/ClaimingSponsoredClaimable.tsx:95]: is `delay` the same as `setTimeout`? if so can we stop using lodash?

**Review by greg-schrammel — APPROVED** [review]

File: src/screens/claimables/ClaimingSponsoredClaimable.tsx (hunk 3 above)
Comment (benisgold) [reply src/screens/claimables/ClaimingSponsoredClaimable.tsx:95]: why?

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@benisgold/claim-caching-fixes/3f21063dfd9df39bd3b4574f8c207661ecfa57ed.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@benisgold/claim-caching-fixes/3f21063dfd9df39bd3b4574f8c207661ecfa57ed.ipa) for 3f21063dfd9df39bd3b4574f8c207661ecfa57ed

**Pull request merged**